control/plugin_manager.go

```diff
@@ -355,10 +355,45 @@ func (p *pluginManager) LoadPlugin(details *pluginDetails, emitter gomit.Emitter
 	lPlugin.ConfigPolicy = cp
 
 	if resp.Type == plugin.CollectorPluginType {
+		cfgNode := p.pluginConfig.getPluginConfigDataNode(core.PluginType(resp.Type), resp.Meta.Name, resp.Meta.Version)
+
+		if lPlugin.ConfigPolicy != nil {
+			// Get config policy from root/plugin defaults
+			ncd := lPlugin.ConfigPolicy.Get([]string{""})
+			_, errs := ncd.Process(cfgNode.Table())
+			if errs != nil && errs.HasErrors() {
+				fields := make(map[string]interface{})
+				se := serror.New(fmt.Errorf("ConfigPolicy processing failed. Not loading collector:%v:%v", resp.Meta.Name, resp.Meta.Version))
+				for i, e := range errs.Errors() {
+					idx := fmt.Sprintf("%d", i)
+					fields[idx] = e.Error()
+				}
+				fields["name"] = resp.Meta.Name
+				fields["version"] = resp.Meta.Version
+				se.SetFields(fields)
+				return nil, se
+			}
+
+			// Update config policy with defaults
+			cp, err = c.GetConfigPolicy()
```

Comment:
Is this block needed given https://github.com/tiffanyfj/snap/blob/00a75a229df3f72ef5a4f5c41865a55699481945/control/plugin_manager.go#L343-L355